umh-core/pkg/service/protocolconverter/protocolconverter.go

```diff
@@ -0,0 +1,756 @@
+
+	return nil
+}
+
+// UpdateInManager modifies an existing protocolconverter configuration.
+func (p *ProtocolConverterService) UpdateInManager(
+	ctx context.Context,
+	filesystemService filesystem.Service,
+	cfg *protocolconverterserviceconfig.ProtocolConverterServiceConfig,
+	protConvName string,
+) error {
+	if p.connectionManager == nil {
+		return errors.New("connection manager not initialized")
+	}
+	if p.dataflowComponentManager == nil {
+		return errors.New("dataflowcomponent manager not initialized")
+	}
+
+	p.logger.Infof("Updating protocolconverter %s", protConvName)
+
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	underlyingName := p.getUnderlyingName(protConvName)
+
+	// Check if the connectionconfig exists
+	foundConn := false
+	indexConn := -1
+	for i, config := range p.connectionConfig {
+		if config.Name == underlyingName {
+			foundConn = true
+			indexConn = i
+			break
+		}
+	}
+	// Check if the dfcconfig exists
+	foundDFC := false
+	indexDFC := -1
+	for i, config := range p.dataflowComponentConfig {
+		if config.Name == underlyingName {
+			foundDFC = true
+			indexDFC = i
+			break
+		}
+	}
+
+	if !foundConn || !foundDFC {
+		return ErrServiceNotExist
+	}
+
+	// Convert our connection config to Connection service config
+	connConfig, dfcConfig, err := p.GenerateConfig(cfg, protConvName)
+	if err != nil {
+		return fmt.Errorf("failed to generate configs: %w", err)
+	}
+
+	// Create a config.ConnectionConfig that wraps the ConnectionServiceConfig
+	connCurrentDesiredState := p.connectionConfig[indexConn].DesiredFSMState
+	p.connectionConfig[indexConn] = config.ConnectionConfig{
+		FSMInstanceConfig: config.FSMInstanceConfig{
+			Name:            underlyingName,
+			DesiredFSMState: connCurrentDesiredState,
+		},
+		ConnectionServiceConfig: connConfig,
+	}
+
+	// Create a config.DataflowComponentConfig that wraps the DataflowComponentServiceConfig
+	dfcCurrentDesiredState := p.dataflowComponentConfig[indexDFC].DesiredFSMState
+	p.dataflowComponentConfig[indexDFC] = config.DataFlowComponentConfig{
+		FSMInstanceConfig: config.FSMInstanceConfig{
+			Name:            underlyingName,
+			DesiredFSMState: dfcCurrentDesiredState,
+		},
+		DataFlowComponentServiceConfig: dfcConfig,
+	}
+
+	return nil
+}
+
+// RemoveFromManager deletes a protocolconverter configuration.
+func (p *ProtocolConverterService) RemoveFromManager(
+	ctx context.Context,
+	filesystemService filesystem.Service,
+	protConvName string,
+) error {
+	if p.connectionManager == nil {
+		return errors.New("connection manager not initialized")
+	}
+
+	if p.dataflowComponentManager == nil {
+		return errors.New("dataflowcomponent manager not initialized")
+	}
+
+	p.logger.Infof("Removing dataflowcomponent %s", protConvName)
+
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	underlyingName := p.getUnderlyingName(protConvName)
+
+	connSliceRemoveByName := func(in []config.ConnectionConfig, name string) []config.ConnectionConfig {
```

Comment:
It could make sense to give all of our configs an implementation of an interface to return Name.
This would allow us to have a generic filter function.

Reply:
also thought about that, but decided not to use generics (as they are a more complicated pattern and it is not really much duplicated code)